src/pages/collectProcess.js

```diff
@@ -70,13 +70,11 @@ export const tubeCollectedTemplate = (participantData, biospecimenData) => {
                     siteTubesList?.forEach((obj, index) => {
                         const notCollectedOptions = siteTubesList.filter(tube => tube.concept === obj.concept)[0].tubeNotCollectedOptions;
                         const deviationOptions = siteTubesList.filter(tube => tube.concept === obj.concept)[0].deviationOptions;
-
-                        const tubeCollected = (biospecimenData[obj.concept][conceptIds.collection.tube.isCollected] === conceptIds.yes);
-                        const tubeDeviated = (biospecimenData[obj.concept][conceptIds.collection.tube.isDeviated] === conceptIds.yes);
+                        const tubeCollected = (biospecimenData[obj.concept]?.[conceptIds.collection.tube.isCollected] === conceptIds.yes);
+                        const tubeDeviated = (biospecimenData[obj.concept]?.[conceptIds.collection.tube.isDeviated] === conceptIds.yes);
 
                         let required = false;
-
-                        if (biospecimenData[obj.concept] && biospecimenData[obj.concept][conceptIds.collection.tube.isCollected] !== conceptIds.no) { 
+                        if (biospecimenData[obj.concept] && biospecimenData[obj.concept]?.[conceptIds.collection.tube.isCollected] !== conceptIds.no) { 
```

Comment:
The conditional chaining isn't strictly needed here because if the property is missing the first part of the if will short circuit the check.  The question here is do we want to do it for consistency? Simply `biospecimenData[obj.concept]?.[conceptIds.collection.tube.isCollected] !== conceptIds.no` without the `biospecimenData[obj.concept] && ` would work

Reply:
Agree and updated. Thanks.